src/op_analytics/cli/subcommands/pulls/defillama.py

```diff
@@ -22,50 +24,70 @@
 METADATA_TABLE = "defillama_stablecoins_metadata"
 
 
-def get_data(session, url):
-    start = time.time()
-    resp = session.request(
-        method="GET",
-        url=url,
-        headers={"Content-Type": "application/json"},
-    ).json()
-    log.info(f"Fetched from {url}: {time.time() - start:.2f} seconds")
-    return resp
-
-
-def process_breakdown_stables(data):
-    peg_type = data["pegType"]
-    balances = data["chainBalances"]
+def get_data(session, url: str) -> Optional[dict]:
+    """
+    Fetches data from a given URL using the provided session.
 
-    rows = []
-    for chain, balance in balances.items():
-        tokens = balance["tokens"]
+    Args:
+        session: The HTTP session for making requests.
+        url (str): The URL to fetch data from.
 
-        for datapoint in tokens:
-            row = {}
-            row["chain"] = chain
-            row["dt"] = dt_fromepoch(datapoint["date"])
+    Returns:
+        Optional[dict]: The JSON response as a dictionary, or None if an error occurred.
+    """
+    start = time.time()
+    try:
+        response = session.get(url, headers={"Content-Type": "application/json"})
+        response.raise_for_status()
+        resp_json = response.json()
+        log.info(f"Fetched from {url}: {time.time() - start:.2f} seconds")
+        return resp_json
+    except Exception as e:
+        log.error(f"Failed to fetch data from {url}: {e}")
+        return None
+
+
+def process_breakdown_stables(
+    data: dict, days: int = 30
+) -> Tuple[pl.DataFrame, Dict[str, Optional[str]]]:
+    """
+    Processes breakdown data for stablecoins, filtering for the most recent N days.
 
-            try:
-                row["circulating"] = datapoint["circulating"][peg_type]
-            except KeyError:
-                row["circulating"] = None
+    Args:
+        data (dict): The breakdown data from the API.
+        days (int): The number of days to filter in the processed data.
 
-            try:
-                row["bridged_to"] = datapoint["bridgedTo"][peg_type]
-            except KeyError:
-                row["bridged_to"] = None
+    Returns:
+        Tuple[pl.DataFrame, Dict[str, Optional[str]]]: A Polars DataFrame with breakdown data and metadata dictionary.
```

Comment:
Since the function has type hints perhaps it would be ok to leave out types from docstrings for less redundancy.